src/commands/synthetics/command.ts

```diff
@@ -119,12 +119,17 @@ export class RunTestCommand extends Command {
 
     // Rendering the results.
     this.reporter?.reportStart({startTime})
+
     const locationNames = triggers.locations.reduce((mapping, location) => {
       mapping[location.id] = location.display_name
 
       return mapping
     }, {} as LocationsMapping)
+
     let hasSucceeded = true // Determine if all the tests have succeeded
+
+    const testResultsAlreadyDisplayed = new Set<string>()
```

Comment:
If I understand correctly the issue is that we iterate on `tests` which will contain multiple times the same test. 

But here we'll be iterating multiple times on the same results (`const testResults = results[test.public_id]`) for each override and we'll be updating the summary multiple times for the same results. One way we could fix that is just by iterating on `results`.

However with this interface no way to properly retrieve the `executionRule` of the override associated with the test run and fail only when we are supposed to 🤔 
- 2 overrides, one blocking other non-blocking
- only the non-blocking result fail
- `hasSucceeded` would be set to false I believe?

One way we could fix this in a non-breaking way could be to add a reference to the `Test` override associated with each `PollResult` (`PollResult.test: Test`) and start ignoring `tests` in this reporter, wdyt?

Reply:
I did not anticipate the amount of changes we had to make in order to take into account this case. I took this opportunity to write some Jest tests (~70% of the changes). I will try to create separate pull-requests to ease the review, or group the set of changes in commits, on the same PR.